applications/app/controllers/InteractiveController.scala

```diff
@@ -68,121 +64,90 @@ class InteractiveController(
-    val htmlResponse = () => InteractiveHtmlPage.html(model)
-    val jsonResponse = () => views.html.fragments.interactiveBody(model)
-    renderFormat(htmlResponse, jsonResponse, model, Switches.all)
   }
 
   override def canRender(i: ItemResponse): Boolean = i.content.exists(_.isInteractive)
 
-  def renderItemLegacy(item: ItemResponse)(implicit request: RequestHeader): Future[Result] = {
-    itemResponseToModel(Future.successful(item)) map {
-      case Left((model, _)) => render(model)
-      case Right(other)     => RenderOtherStatus(other)
-    }
+  def renderNonDCR(page: InteractivePage)(implicit request: RequestHeader): Future[Result] = {
+    val htmlResponse = () => InteractiveHtmlPage.html(page)
+    val jsonResponse = () => views.html.fragments.interactiveBody(page)
+    val res = renderFormat(htmlResponse, jsonResponse, page, Switches.all)
+    Future.successful(res)
   }
 
-  def renderDCR(item: ItemResponse)(implicit request: RequestHeader): Future[Result] = {
-    itemResponseToModel(Future.successful(item)) flatMap {
-      case Left((model, blocks)) => {
-        val pageType = PageType.apply(model, request, context)
-        remoteRenderer.getInteractive(wsClient, model, blocks, pageType)
-      }
-      case Right(other) => Future.successful(RenderOtherStatus(other))
-    }
+  def renderHtml(page: InteractivePage, blocks: Blocks)(implicit request: RequestHeader): Future[Result] = {
+    val pageType = PageType.apply(page, request, context)
+    remoteRenderer.getInteractive(wsClient, page, blocks, pageType)
   }
 
-  def renderDCRJson(item: ItemResponse)(implicit request: RequestHeader): Future[Result] = {
-    itemResponseToModel(Future.successful(item)) map {
-      case Left((model, blocks)) => {
-        val data =
-          DotcomRenderingDataModel.forInteractive(model, blocks, request, PageType.apply(model, request, context))
-        val dataJson = DotcomRenderingDataModel.toJson(data)
-        common.renderJson(dataJson, model).as("application/json")
-      }
-      case Right(other) => RenderOtherStatus(other)
-    }
+  def renderJson(page: InteractivePage, blocks: Blocks)(implicit request: RequestHeader): Future[Result] = {
+    val data =
+      DotcomRenderingDataModel.forInteractive(page, blocks, request, PageType.apply(page, request, context))
+    val dataJson = DotcomRenderingDataModel.toJson(data)
+    val res = common.renderJson(dataJson, page).as("application/json")
+    Future.successful(res)
   }
 
-  def renderDCRAmp(item: ItemResponse)(implicit request: RequestHeader): Future[Result] = {
-    itemResponseToModel(Future.successful(item)) flatMap {
-      case Left((model, blocks)) => {
-        val pageType = PageType.apply(model, request, context)
-        remoteRenderer.getAMPInteractive(wsClient, model, blocks, pageType)
-      }
-      case Right(other) => Future.successful(RenderOtherStatus(other))
-    }
-  }
-
-  def itemResponseToInteractivePickerInputData(item: ItemResponse)(implicit
```

Comment:
I simplified and inlined this code into `renderItem` so the standalone method was no longer needed.